pkg/runs/junitReporter_test.go

```diff
@@ -76,3 +254,188 @@ func TestJunitReportWorks(t *testing.T) {
+	if err != nil {
+		assert.Fail(t, "junit report does not exist in correct place. "+err.Error())
+	}
+	assert.True(t, isExists, "Junit report does not exist in the correct place.")
+
+	// We expect a report like this:
+	expectedReport := `<?xml version="1.0" encoding="UTF-8" ?>
+	<testsuites id="myGroup" name="Galasa test run" tests="4" failures="1" time="0">
+		<testsuite id="myTestRun1" name="myStream1/myBundle1/com.myco.MyClass1" tests="2" failures="1" time="0">
+			<testcase id="method1.1" name="method1.1" time="0"></testcase>
+			<testcase id="method1.2" name="method1.2" time="0">
+				<failure message="Failure messages are unavailable at this time" type="Unknown"></failure>
+			</testcase>
+		</testsuite>
+		<testsuite id="myTestRun2" name="myStream2/myBundle2/com.myco.MyClass2" tests="2" failures="0" time="0">
+			<testcase id="method2.1" name="method2.1" time="0"></testcase>
+			<testcase id="method2.2" name="method2.2" time="0"></testcase>
+		</testsuite>
+	</testsuites>`
+
+	actualContents, err := mockFileSystem.ReadTextFile("myReportJunitFilename")
+	if err != nil {
+		assert.Fail(t, "Could not read the junit file. "+err.Error())
+	}
+
+	expected := stripWhitespace(expectedReport)
+	actual := stripWhitespace(actualContents)
+
+	assert.Equal(t, len(expected), len(actual), "Lengths are not valid. Expected:%d Actual:%d", len(expected), len(actual))
+
+	for index := range expected {
+		expectedChar := expected[index]
+		actualChar := actual[index]
+		expectedCharOrd, _ := strconv.Atoi(string(expectedChar))
+		actualCharOrd, _ := strconv.Atoi(string(actualChar))
+		assert.Equal(t, expectedChar, actualChar, "Characters are not the same! expected:'%v' actual:'%v'", expectedCharOrd, actualCharOrd)
+	}
+
+	assert.EqualValues(t, actual, expected)
+}
+
+func TestJunitReportWithManyRunsAndMixedResultTestsReturnsAlphabeticallyOk(t *testing.T) {
+	// Given...
+	mockFileSystem := files.NewMockFileSystem()
+
+	finishedRuns1 := TestRun{
+		Name:      "zoo",
+		Bundle:    "myBundle1",
+		Class:     "com.myco.MyClass1",
+		Stream:    "myStream1",
+		Status:    "myStatus1",
+		Result:    "FAILED",
+		Overrides: make(map[string]string, 1),
+		Tests:     []TestMethod{{Method: "method1.1", Result: "Passed"}, {Method: "method1.2", Result: "failed"}}}
+
+	finishedRuns2 := TestRun{
+		Name:      "myTestRun2",
+		Bundle:    "myBundle2",
+		Class:     "com.myco.MyClass2",
+		Stream:    "myStream2",
+		Status:    "myStatus2",
+		Result:    "PASSED",
+		Overrides: make(map[string]string, 1),
+		Tests:     []TestMethod{{Method: "method2.1", Result: "Passed"}, {Method: "method2.2", Result: "Passed"}}}
+
+	finishedRuns3 := TestRun{
+		Name:      "eagle",
+		Bundle:    "myBundle2",
+		Class:     "com.myco.MyClass2",
+		Stream:    "myStream2",
+		Status:    "myStatus2",
+		Result:    "PASSED",
+		Overrides: make(map[string]string, 1),
+		Tests:     []TestMethod{{Method: "method3.1", Result: "Passed"}}}
+
+	finishedRunsMap := make(map[string]*TestRun, 1)
```

Comment:
Seems to me that all your tests are the same structure, just with different input and expected strings ?
If that's the case, then one copy of the logic should be good enough, and each test just uses that sub-function ?